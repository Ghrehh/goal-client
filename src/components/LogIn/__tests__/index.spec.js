import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { mount } from 'enzyme';
import { LogIn } from '../';

describe('LogIn', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.setAuthMock = jest.fn();
    testContext.createSessionMock = jest.fn();

    testContext.component = (
      <LogIn
        setAuth={testContext.setAuthMock}
        createSession={testContext.createSessionMock}
        loading={false}
        error={undefined}
      />
    );
  });

  describe('snapshot', () => {
    beforeEach(() => {
      const renderer = new ShallowRenderer();
      testContext.result = renderer.render(testContext.component);
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });

  describe('submitting form', () => {
    beforeEach(() => {
      const wrapper = mount(testContext.component);
      wrapper.find('.name').instance().value = 'name';
      wrapper.find('.password').instance().value = 'password';
      wrapper.find('button').simulate('click');
    });

    it('calls the create session mock with the correct arguments', () => {
      expect(testContext.createSessionMock).toBeCalledWith({
        variables: { name: 'name', password: 'password' }
      });
    });
  });

  describe('saving auth on successful log in', () => {
    beforeEach(() => {
      const wrapper = mount(testContext.component);

      wrapper.setProps({
        setAuth: testContext.setAuthMock,
        createSession: testContext.createSessionMock,
        loading: false,
        rememberToken: 'rememberToken',
        userId: 'userId',
        error: undefined
      })
    });

    it('calls the create session mock with the correct arguments', () => {
      expect(testContext.setAuthMock).toBeCalledWith({
        rememberToken: 'rememberToken',
        userId: 'userId'
      });
    });
  });
});
