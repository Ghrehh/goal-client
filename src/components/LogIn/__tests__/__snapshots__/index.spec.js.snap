// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn snapshot renders the component correctly 1`] = `
<LoadingAndErrorHandler
  loading={false}
>
  <form>
    <input
      className="name"
    />
    <input
      className="password"
      type="password"
    />
    <button
      onClick={[Function]}
    >
      Log In
    </button>
  </form>
</LoadingAndErrorHandler>
`;
